code/modules/xenoarchaeology/xenoartifact_labeler.dm

```diff
@@ -0,0 +1,217 @@
+    if(!P.afterattack(target, user)) //In the circumstance the sticker fails, usually means you're doing something you shouldn't be
+        qdel(P)
+
+/obj/item/xenoartifact_labeler/proc/create_label(new_name)
+    var/obj/item/xenoartifact_label/P = new(get_turf(src))
+    if(new_name)
+        P.name = new_name
+        P.set_name = TRUE
+    P.trait_list = sticker_traits
+    P.info = activator+minor_trait+major_trait
+    return P
+
+/obj/item/xenoartifact_labeler/proc/trait_toggle(action, toggle_type, var/list/trait_list, var/list/active_trait_list)
+    var/datum/xenoartifact_trait/description_holder
+    var/new_trait
+    for(var/T in trait_list)
+        new_trait = desc2datum(T)
+        description_holder = new new_trait
+        if(action == "assign_[toggle_type]_[T]")
+            if(!look_for(active_trait_list, T))
+                active_trait_list += list(T)
+                info_list += description_holder.label_desc
+                sticker_traits += new_trait
+            else
+                active_trait_list -= list(T)
+                info_list -= description_holder.label_desc
+                sticker_traits -= new_trait
+
+/obj/item/xenoartifact_labeler/proc/desc2datum(udesc) //This is just a hacky way of getting the info from a datum using its desc becuase I wrote this last and it's not heartbreaking
+    for(var/T in typesof(/datum/xenoartifact_trait))
+        var/datum/xenoartifact_trait/X = new T
+        if((udesc == X.desc)||(udesc == X.label_name))
+            return T
+    return "[udesc]: There's no known information on [udesc]!."
+
+// Not to be confused with labeler
+/obj/item/xenoartifact_label
+    icon = 'icons/obj/xenoarchaeology/xenoartifact_sticker.dmi'
+    icon_state = "sticker_star"
+    name = "Xenoartifact Label"
+    desc = "An adhesive label describing the characteristics of a Xenoartifact."
+    var/info = "" 
+    var/set_name = FALSE
+
+    var/mutable_appearance/sticker_overlay
+
+    var/list/trait_list = list() //List of traits used to compare and generate modifier.
+
+/obj/item/xenoartifact_label/Initialize()
+    icon_state = "sticker_[pick("star", "box", "tri", "round")]"
+    var/sticker_state = "[icon_state]_small"
+    sticker_overlay = mutable_appearance(icon, sticker_state)
+    sticker_overlay.layer = FLOAT_LAYER
+    sticker_overlay.appearance_flags = RESET_ALPHA
+    sticker_overlay.appearance_flags = RESET_COLOR
+    ..()
+    
+/obj/item/xenoartifact_label/afterattack(atom/target, mob/user)
+    for(var/obj/item/xenoartifact_label/L in target.contents)
+        target.name = name //You can update the name but, you should only really get one chance to slueth the traits
+        return FALSE
+    if(istype(target, /mob/living))
+        to_chat(target, "<span class='notice'>[user] sticks a [src] to you.</span>")
+        add_sticker(target)
+        addtimer(CALLBACK(src, .proc/remove_sticker, target), 15 SECONDS)
+        return TRUE
```

Comment:
I don't like the idea of instant stickers on people, since you could put a C4 on a sticker, instantly plant it on someone and theres nothing they can do about it.